clientconn.go

```diff
@@ -966,6 +966,11 @@ func (ac *addrConn) resetTransport() {
 		}
 
 		ac.mu.Lock()
+		if ac.state == connectivity.Shutdown {
```

Comment:
Why add this block? It seems like very code happens (and no i/o) until the next shutdown check in tryAll?

Reply:
This is just being paranoid.
Ideally, after taking `ac.mu`, the first thing should be checking `stats==Shutdown` so we can be sure whatever we read in `ac` is sane.
An made-up situation (not the case today) is, if we somehow set `ac.addrs = nil` when closing addrConns. Then the for loop in tryAdd won't execute, and we will not detect `Shutdown`.

I almost wanted to do the following so we don't need the manual checks everytime. But it needs a lambda...

```go
func (ac *addrConn) withLock(realWork func()) {
  ac.mu.Lock()
  defer ac.mu.Unlock()
  if ac.state == Shutdown {
    return
  }
  realWork()
}
```

Reply:
Ahh gotcha. Ack, your call about remove or keep, just thought I'd mention.